### ISSUE TITLE
Wrong behavior with comparing a boolean index fields

### ISSUE THREAD

**Issue by Yuzal** [ravendb/ravendb#12714]
```
async Task Main()
{
	var result = await session.Query<Index.Result, Index>()
					  		   .Select(e => new
							   {
					  		  	 Index_IsValid = e.IsValid, // returns true,
								 Index_IsValid2 = e.IsValid == true, // returns false but why?
	
								 Entity_IsDifferent = e.IsDifferent, // returns true,
								 Entity_IsDifferent2 = e.IsDifferent == true, // returns true
							   })
							   .ToListAsync(cancellationToken);
}

public class Entity
{ 
	public bool IsDifferent = true;
}

public class Index : AbstractIndexCreationTask<Entity, Index.Result>
{
	public class Result : Entity
	{
		public bool IsValid;
		
		public Result()
		{
			Map = results => from e in results
							 select new
							 {
							 	IsValid = true,
							 };
		}

		Store(e => e.IsValid, FieldStorage.Yes);
		
	}
}
```

I dont understand why i cant compare my boolean field from an index with a bool while it works with a normal database entity?
Did i mess up something with my index?

**Comment by ayende** [comment]
I assume that you have the `IsDifferent` field on the entity?

The issue is that your `IsValid` field is stored in the index. When you return that to the user, you are getting the following result:

```
{
    "IsValid": "true",
    "IsValid2": false,
}
```

In other words, the stored field is stored as a _string_.
However, for `Index_IsValid2`,  you are comparing it server side, where you compare `"true" == true`. As you can imagine, that isn't a true statement.

Basically, this is a type issue when you lose the boolean in favor of a string value.

**Comment by Yuzal** [comment]
> I assume that you have the IsDifferent field on the entity?

yes

soo how can i compare this field correctly in c#? 
something like `e.IsValid == false`

and more important... how can i avoid that other devs quering/selecting these kind of index having the same trouble? 
is there some option to define the type of the stored field while comparing?

**Comment by ayende** [comment]
You have to take that into account in your queries, or set the the `ProjectionBehavior` to force it

See: https://ravendb.net/docs/article-page/5.2/Csharp/client-api/session/querying/how-to-project-query-results#projection-behavior

I would avoid doing those sort of things at the query time. There is no advantage for doing that on the server side vs. client side.
